src/frontends/tensorflow/include/openvino/frontend/tensorflow/node_context.hpp

```diff
@@ -68,7 +68,27 @@ class NodeContext : public ov::frontend::NodeContext {
     const OutputVector& m_inputs;
 };
 
-using CreatorFunction = std::function<ov::OutputVector(const ov::frontend::tensorflow::NodeContext&)>;
+using CreatorFunctionIndexed = std::function<ov::OutputVector(const ov::frontend::tensorflow::NodeContext&)>;
+using CreatorFunctionNamedAndIndexed = std::function<NamedOutputVector(const ov::frontend::tensorflow::NodeContext&)>;
+
+struct CreatorFunction {
+
+    CreatorFunction() = default;
+    CreatorFunction(CreatorFunctionIndexed _func): func(_func) {}
+    CreatorFunction(CreatorFunctionNamedAndIndexed _func): func(_func) {}
+
+    NamedOutputVector operator() (const ov::frontend::tensorflow::NodeContext& node) const {
+        if(func.is<CreatorFunctionIndexed>()) {
+            auto outputs = func.as<CreatorFunctionIndexed>()(node);
+            return NamedOutputVector(outputs.begin(), outputs.end());
+        } else {
+            return func.as<CreatorFunctionNamedAndIndexed>()(node);
+        }
+    }
+
+    Any func;
```

Comment:
Any is excessive here
We can create two fields corresponding to each creator function

Reply:
@slyalin do you agree? I think you were in favor of this change

Reply:
Any is template class and it blow up the binary size.
So, it's usage should be justified.

In your example we have selection between 2 types, so IMO Any is excessive here

Reply:
Yes I voted for Any because it exactly provided what we needed here without introducing another Union-like class. As we don't have std::variant from c++ 17, ov::Any always pops up as an alternative. But it looks like ov::Any specializes too many functions underneath, that means it is not well designed to be a general purpose tool.

While I tend to share this expectation about binary size increase, @ilya-lavrenov, do you have any measurements that can show that each extra type specialized in Any gives us the increase in binary size not adequate to the task we are trying to solve?

Reply:
Usage of ov::Any does not provide simplification of the code here. 
You still have to check for specific function type using `is<T>()` method:
```c++
if (func_a != nullptr)
  func_a(arguments);
else if (func_b != nullptr)
  func_b(arguments);
```

ov::Any is designed to bypass / store different types of values in the API to reduce a number of overloaded methods. While in your example you use ov::Any as an alternative of:
```
FuncA func_a;
FuncB func_b;
```